plugins/tcpinfo/tcpinfo.cc

```diff
@@ -360,36 +361,37 @@ TSPluginInit(int argc, const char *argv[])
     case 'e':
       i = strtoul(optarg, &endptr, 10);
       if (*endptr != '\0' || i > 3) {
-        TSError("[tcpinfo invalid rolling-enabled argument, using default of %d, valid values are 0-3", config->rolling_enabled);
+        TSError("[tcpinfo invalid rolling-enabled argument, '%s', using default of %d, valid values are 0-3", optarg,
+                rolling_enabled);
       } else {
-        config->rolling_enabled = i;
+        rolling_enabled = i;
       }
       break;
     case 'H':
       i = strtoul(optarg, &endptr, 10);
       if (*endptr != '\0' || i > 23) {
-        TSError("[tcpinfo invalid rolling-offset-hr argument, using default of %d, valid values are 0-23",
-                config->rolling_offset_hr);
+        TSError("[tcpinfo invalid rolling-offset-hr argument, '%s', using default of %d, valid values are 0-23", optarg,
+                rolling_offset_hr);
       } else {
-        config->rolling_offset_hr = i;
+        rolling_offset_hr = i;
       }
       break;
     case 'S':
       i = strtoul(optarg, &endptr, 10);
       if (*endptr != '\0' || i < 60 || i > 86400) {
-        TSError("[tcpinfo invalid rolling-interval-sec argument, using default of %d, valid values are 60-86400",
-                config->rolling_interval_sec);
+        TSError("[tcpinfo invalid rolling-interval-sec argument, '%s', using default of %d, valid values are 60-86400", optarg,
+                rolling_interval_sec);
       } else {
-        config->rolling_interval_sec = i;
+        rolling_interval_sec = i;
       }
       break;
     case 'M':
-      i = strtoul(optarg, &endptr, 10);
-      if (*endptr != '\0' || i < 10) {
-        TSError("[tcpinfo invalid rolling-size-mb argument, using default of %d, values must be 10 or greater",
-                config->rolling_interval_sec);
+      i = ink_atoui(optarg);
```

Comment:
Now that you are using `ink_atoui()`, this option should just be `--rolling-size`, since the operator would do `--rolling-size=10M`.